### DISCUSSION TITLE
How to disable mason-lspconfig autoinstall ruff_lsp?

### DISCUSSION THREAD

**Discussion by baoyuxu** [LazyVim/LazyVim#3907] [Q&A] [answered]
<img width="513" alt="image" src="https://github.com/LazyVim/LazyVim/assets/18047650/c7692a43-1945-4f8a-98f0-8f089f23f474">

I installed pyright and ruff_lsp and then uninstall ruff_lsp using mason. But everytime I open a .py file, it will autoinstall ruff_lsp. Since I only want to use pyright, how to disable this autoinstall?

**Comment by dpetka2001** [comment, marked as answer]
It's in the [docs](https://www.lazyvim.org/plugins/lsp#nvim-lspconfig). You set `mason = false` for the server you don't want to auto-install via Mason.

**Reply by baoyuxu** [reply]
Thanks for your reply, it works! However I got new error，Do you know how to deal with this? 
<img width="1196" alt="image" src="https://github.com/LazyVim/LazyVim/assets/18047650/4b0e0718-365a-44f3-8a5c-5096b6d87ea7">

**Reply by dpetka2001** [reply]
I think you should also add `enabled = false` for that error message.

**Reply by tvannahl** [reply]
Thank you @baoyuxu for asking the question and @dpetka2001 for the help. That was driving me crazy.

I've created the file `~/.config/nvim/lua/plugins/no-ruff-lsp.lua`:

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        ruff_lsp = {
          mason = false,
          enabled = false,
        },
      },
    },
}
```

**Reply by RBloemers** [reply]
<3 I had same issues with jsonls and lua_ls on freebsd, drove me nuts those popups on startup, finally its silenced \o/
